src/data/real/sign.lean

```diff
@@ -0,0 +1,86 @@
+/-
+Copyright (c) 2021 Kexing Ying. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Kexing Ying
+-/
+import data.real.basic
+
+/-!
+# Real sign function
+
+This file introduces and contains some results about `real.sign` which maps negative
+real numbers to -1 and nonnegative numbers to 1.
+
+## Main definitions
+
+ * `real.sign r` is -1 if `r < 0` and 1 otherwise
+
+## Tags
+
+sign function
+-/
+
+namespace real
+
+/-- The sign function that maps negative real numbers to -1 and nonnegative numbers to 1. -/
+noncomputable
+def sign (r : ℝ) : ℝ := if r < 0 then -1 else 1
+
+lemma sign_neg {r : ℝ} (hr : r < 0) : sign r = -1 :=
```

Comment:
```suggestion
lemma sign_of_neg {r : ℝ} (hr : r < 0) : sign r = -1 :=
```
`sign_neg` would be `sign (-r) = -sign r` for `r \ne 0`. Also a useful lemma, but maybe for another PR.

Reply:
I will add it here since its pretty short